gomock/matchers.go

```diff
@@ -382,6 +410,22 @@ func Not(x any) Matcher {
 	return notMatcher{Eq(x)}
 }
 
+// Regex checks whether parameter matches the associated regex.
+//
+// Example usage:
+//
+// Regex("[0-9]{2}:[0-9]{2}").Matches("23:02") // returns true
+// Regex("[0-9]{2}:[0-9]{2}").Matches([]byte{50,51,58,48,50}) // returns true
+// Regex("[0-9]{2}:[0-9]{2}").Matches("hello world") // returns false
+// Regex("[0-9]{2}").Matches(21) // returns false as it's not a valid type
+func Regex(regexStr string) Matcher {
+	compiledRegexp, err := regexp.Compile(regexStr)
+	if err == nil {
+		return regexMatcher{regex: compiledRegexp}
+	}
+	return regexMatcher{regex: nil, compileErr: err}
```

Comment:
```suggestion
	return regexMatcher{regex: regexp.MustCompile(regexStr)}
```
_(This suggested change must be accepted along with the one about that changes the definition of `regexMatcher`)_